File: java-springboot/src/main/java/com/example/demo/controllers/UserController.java
package com.example.demo.controllers;

import com.example.demo.models.UserModel;
import com.example.demo.services.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequiredArgsConstructor
@RequestMapping("/")
public class UserController {
    private final UserService userService;

    @GetMapping("/health")
    @ResponseStatus(HttpStatus.OK)
    public void health() {
    }

    @GetMapping("/users")
    public List<UserModel> listUsers() {
        return userService.listUsers();
    }

    @GetMapping("/users/example")
    public UserModel getExampleUser() {
        return userService.listUsers().getFirst();
    }
}

File: node-express/src/controllers/userController.js
const userService = require('../services/userService');

function listUsers(req, res) {
  const users = userService.listUsers();

  return res.json(users);
}

function getExampleUser(req, res) {
  const user = userService.getExampleUser();

  return res.json(user);
}

module.exports = {
  listUsers,
  getExampleUser
};
